protobuf/full_snapshot.proto

```diff
@@ -625,3 +629,37 @@ message CustomTypeInformation {
     string type = 2;
   }
 }
+
+message QueryPlanInformation {
+  int32 query_idx = 1;
+  int64 plan_id = 2;
+  string explain_plan = 3;
+  google.protobuf.Timestamp plan_captured_time = 4;
+}
+
+message QueryPlanStatistic {
+  int32 query_idx = 1;
+  int64 plan_id = 2;
```

Comment:
It may be useful to instead point to a `plan_idx`, and follow our standard reference/information/stats logic in the snapshot - that should also make the server-side processing easier, I believe.